app/src/components/datetime/TimeRangeSelector.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { PropsWithChildren, useState } from "react";
+import { Dialog, DialogTrigger } from "react-aria-components";
+import { css } from "@emotion/react";
+
+import {
+  Button,
+  Flex,
+  Heading,
+  Icon,
+  Icons,
+  ListBox,
+  ListBoxItem,
+  Popover,
+  PopoverArrow,
+  TimeRangeForm,
+  View,
+} from "@phoenix/components";
+import { timeRangeFormatter } from "@phoenix/utils/timeFormatUtils";
+
+import { LAST_N_TIME_RANGES } from "./constants";
+import { OpenTimeRangeWithKey } from "./types";
+import { getTimeRangeFromLastNTimeRangeKey, isTimeRangeKey } from "./utils";
+
+export type TimeRangeSelectorProps = {
+  isDisabled?: boolean;
+  value: OpenTimeRangeWithKey;
+  onChange: (timeRange: OpenTimeRangeWithKey) => void;
+};
+
+const listBoxCSS = css`
+  width: 120px;
+  padding: var(--ac-global-dimension-size-50);
+`;
+
+/**
+ * Get the display text for the time range key. Shows the explicit time range in the case of "custom"
+ */
+function getDisplayText({ timeRangeKey, start, end }: OpenTimeRangeWithKey) {
+  if (timeRangeKey === "custom") {
+    return timeRangeFormatter({ start, end });
+  }
+  const rangeValue = LAST_N_TIME_RANGES.find(
+    (range) => range.key === timeRangeKey
+  );
+  if (!rangeValue) {
+    // Should never happen but must make sure to handle it
+    return "invalid";
+  }
+  return rangeValue.label;
+}
+
+export function TimeRangeSelector(props: TimeRangeSelectorProps) {
+  const { value, isDisabled, onChange } = props;
+  const { timeRangeKey, start, end } = value;
+  const [isOpen, setOpen] = useState<boolean>(false);
+  return (
+    <DialogTrigger>
+      <Button
+        size="S"
+        icon={<Icon svg={<Icons.CalendarOutline />} />}
+        isDisabled={isDisabled}
+        onPress={() => setOpen(true)}
+      >
+        {getDisplayText(value)}
+      </Button>
+      <Popover placement="bottom end" isOpen={isOpen}>
+        <Dialog>
+          <PopoverArrow />
+          <Flex direction="row">
+            <CustomTimeRangeWrap visible={timeRangeKey === "custom"}>
+              {/* We force re-mount to reset the dirty state in the form */}
```

Comment:
Is this still relevant? I don't see any key manipulation that would cause remounts

Reply:
it's that I want the default value to reset in the form, otherwise if you switch back to a lastN time range and come back you end up with the dirty state and I wanted it to reflect the time range you selected.